### PR TITLE
All "Sands" to "Sand"

### LINKED ISSUE #8434: Inconsistent terrain name (Sand or Sands)
Both “Sand” and “Sands” are used to refer to the same terrain type. Because “Hills” and “Mountains” are always plural, and other singular ones are always singular, this is probably just an oversight rather than an intentional word choice.

I checked help and map terrain in game as well as the unit database (picture below), but there may be other places where “Sand” and “Sands” are used interchangeably.

The questions are;
Q1: Should they be “Sand” or “Sands” ?
Q2: Where are the codes for these?

Unfortunately, I have no idea which file has control over which description. If there is a simple fix by an expert here who can filter through all the files and replace all the relevant “Sand” with “Sands” or vice versa, that would be great.

HttT Sand(s)

### PR DESCRIPTION
**Pull request by OrekOrek**
Fixes #8434

### CODE REVIEW

**Comment by stevecotton** [comment]
While 1.18 has branched, I feel it would be good to stay in string-freeze on master a while longer, to make backports easier.

At least for "Beach Sands" and "Desert Sands", the extra "s" sounds right to me (grew up in the UK, specifically East Midlands).

**Comment by CelticMinstrel** [comment]
> At least for "Beach Sands" and "Desert Sands", the extra "s" sounds right to me (grew up in the UK, specifically East Midlands).

I'm not sure which sounds _better_ to me, but without the S doesn't sound _wrong_.

That doesn't mean I've _convinced_ this is the correct thing to do, but it means I'm not opposed to it.

**Comment by Wedge009** [comment]
In the context of the images shown in #8434, 'Sands' still makes sense to me because it's referring to multiple variants of sand, including Desert Sand and Beach Sand.

Other than that, I'm not sure I have strong opinion either way, or even of the value of this change (I've been accused of making similarly trivial proposals to strings in the past, and the attacks from certain individuals are... not very gracious, I shall say.)

**Comment by OrekOrek** [comment]
History is researched:

1. Back in the day, only “real” sand terrains existed such as
```
    id=sand
    name= _ "Sand"
    editor_name= _ "Beach Sands"
```

2. https://github.com/wesnoth/wesnoth/commit/f456850875847479c964171dc3503abb689d81a7#diff-0b3fb4c291132ff1f4c3ec2f0a68a5ce8f01796cb83f48831a38f02a39798efcL289-R358
1.11.8 (2013)
“Virtual” parent terrain (Terrain Archetype) for sand terrains was newly introduced.
Then, `id=sand` was switched to this virtual parent terrain, and `id=sand_beach` was made for real Beach Sands terrain instead.
```
    id=sand
    name= _ "Sand"
    editor_name= _ "Sands"
```
```
    id=sand_beach
    name= _ "Sand"
    editor_name= _ "Beach Sands"
```
In the process, name and editor_name for the virtual parent terrain were made different probably because editor_name was _wrongly_ influenced by that of Beach **_Sands_**. I personally believe that this is the root of the problem.


3. https://github.com/wesnoth/wesnoth/commit/10854d4802c4953cf05a8e28cb81ba5ab8d52b29#diff-0b3fb4c291132ff1f4c3ec2f0a68a5ce8f01796cb83f48831a38f02a39798efcR2229-R2240
1.11.8 (2013)
Terrain Archetype is slightly edited and moved to the current place.

4. Today, among 17 Terrain Archetypes, this is the only one which has a different name and editor_name.
https://github.com/wesnoth/wesnoth/blob/8cebaf5b51838d82525835dc67085ec2c9c81433/data/core/terrain.cfg#L3278-L3279
Archetypes

In my opinion, name and editor_name for “virtual” sand terrain (Terrain Archetype) should be the same so that the description is consistent everywhere.

However, editor_name for 3 “real” terrains [Rocky Sand(s), Desert Sand(s), and Beach Sand(s)] can be decided based on how natural they sound in English because it is not the crux of the issue.

**Comment by Pentarctagon** [comment]
Having the game refer to terrains consistently is important, so I'd lean towards merging this.

File: data/core/terrain.cfg [hunk 1]
```diff
@@ -271,7 +271,7 @@
     symbol_image=flat/sandy-path
     id=road_desert
     name = _ "Road"
-    editor_name= _ "Rocky Sands"
+    editor_name= _ "Rocky Sand"
```
Comment (stevecotton) [inline data/core/terrain.cfg:274]: Wondering if we should rename this completely. The stones aren't organised enough to be called "cobbled", but it's more like a dirt track that's been made more walkable by pushing stones in when it became muddy.

It doesn't look terrible next to desert, but it's notable that UtBS doesn't use this terrain. The only map in mainline that does is EI S04c (Mal Ravanals' Capital).

**Comment by stevecotton** [comment]
Changing "Sands" to "Sand" is easy for the translators, because they're already translating "Sand".

I'd still prefer to leave "Beach Sands" and "Desert Sands" unchanged.

**Comment by Pentarctagon** [comment]
@OrekOrek are you still around to make that tweak?

File: data/core/terrain.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/core/terrain.cfg]: Any issue with it being changed to Gravel now?

**Comment by stevecotton** [comment]
Looks good to me.

**Comment by Pentarctagon** [comment]
Thanks!

**Pull request merged**